flake.nix

```diff
@@ -1,8 +1,37 @@
 {
   description = "Poetry2nix flake";
 
-  outputs = { self }:
+  outputs = { self, nixpkgs }:
+    let
+      # TODO: There must be a better way to provide arch-agnostic flakes..
+      systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs systems (system: f system);
+      # Memoize nixpkgs for different platforms for efficiency.
+      nixpkgsFor = forAllSystems (
+        system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+    in
```

Comment:
You can use flake-utils here: https://github.com/numtide/flake-utils

Reply:
This would be the preferred way since it also comes with a `defaultSystems` implementation that would not require an update from `poetry2nix` side.